test/Extensionsions.Configuration.Object.UnitTests/RecordObjectTest.cs

```diff
@@ -0,0 +1,44 @@
+using FluentAssertions;
+using Microsoft.Extensions.Configuration;
+using Xunit;
+
+namespace Extensions.Configuration.Object.UnitTests
+{
+    public class RecordObjectTest
+    {
+        internal record MyConfiguration
+        {
+            public SubRecordConfiguration SubRecord;
+            public string MyKey;
+        }
+
+        internal record SubRecordConfiguration
+        {
+            public string Title;
+            public string Name;
+            public int Age;
+        }
+
+        [Fact]
+        public void AddObject_WithClassObject_ShouldLoadFieldsIntoConfiguration()
```

Comment:
This could probably be renamed to `AddObject_WithRecordObject_...` not to confuse with the class test